app/model/command/CreateSponsorshipCommand.scala

```diff
@@ -0,0 +1,72 @@
+package model.command
+
+import com.gu.tagmanagement.{SectionEvent, EventType, TagEvent}
+import model._
+import model.command.logic.SponsorshipStatusCalculator
+import org.joda.time.DateTime
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{JsPath, Format}
+import repositories.SponsorshipOperations._
+import repositories.{SponsorshipRepository, Sequences}
+import services.KinesisStreams
+
+case class CreateSponsorshipCommand(
+  validFrom: Option[DateTime],
+  validTo: Option[DateTime],
+  sponsorshipType: String,
+  sponsorName: String,
+  sponsorLogo: Image,
+  sponsorLink: String,
+  tag: Option[Long],
+  section: Option[Long],
+  targeting: Option[SponsorshipTargeting]
+) extends Command {
+
+  override type T = Sponsorship
+
+  override def process()(implicit username: Option[String]): Option[T] = {
+
+    val status = SponsorshipStatusCalculator.calculateStatus(validFrom, validTo)
+
+    val sponsorship = Sponsorship(
+      id = Sequences.sponsorshipId.getNextId,
+      validFrom = validFrom,
+      validTo = validTo,
+      status = status,
+      sponsorshipType = sponsorshipType,
+      sponsorName = sponsorName,
+      sponsorLogo = sponsorLogo,
+      sponsorLink = sponsorLink,
+      tag = tag,
+      section = section,
+      targeting = targeting
+    )
+
+    SponsorshipRepository.updateSponsorship(sponsorship).map { createdSponsorship =>
+
+      if(status == "active") {
```

Comment:
The state is derived from the dates but the marker is needed so that the schedule launcher and expirer know which things to update.